qa/integration/specs/monitoring_api_spec.rb

```diff
@@ -208,7 +208,12 @@
   shared_examples "pipeline metrics" do
     # let(:pipeline_id) { defined?(super()) or fail NotImplementedError }
     let(:settings_overrides) do
-      super().merge({'pipeline.id' => pipeline_id})
+      super().dup.tap do |overrides|
+        overrides['pipeline.id'] = pipeline_id
+        if logstash_service.settings.feature_flag == "persistent_queues"
+          overrides['queue.compression'] = %w(none speed balanced size).sample
```

Comment:
I made the decision to sample the available values instead of exploding the matrix by 4x.